src/utils/utils.rs

```diff
@@ -0,0 +1,23 @@
+use std::{env, fs, os::unix::prelude::PermissionsExt, path::Path};
+
+use crate::error::MiddlewareError;
+
+pub fn create_directory(application_name: &String) -> Result<(), MiddlewareError> {
+    let name = "HOME";
+    match env::var(name) {
+        Ok(v) => println!("{}: {}", name, v),
```

Comment:
println not needed.  Instead use log::debug.  
log::debug!("HOME environment variable is set to {}. Using it", name);